src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/BuildHistoryManager.java

```diff
@@ -50,16 +50,18 @@ public void perform(Job<?, ?> job) throws IOException, InterruptedException {
         // for each completed build...
         do {
             LOG.info("Processing build #" + run.getNumber());
-            for (int i = 0; i < rules.size(); i++) {
-                Rule rule = rules.get(i);
-                LOG.info("Processing rule no " + (i + 1));
-                if (rule.validateConditions(run)) {
-                    LOG.info("Processing actions for rule no " + (i + 1));
-                    rule.performActions(run);
+            if (!run.isKeepLog()) {
```

Comment:
this makes sense when the action is about to delete log but rather not if users wants to delete whole execution or artifacts

I agree but now here - rather proper action - and also provide log when why the log is not deleted

Reply:
from my experience checking `isKeepLog()` is the only way to check with the given api if the user marked a build from the ui as keep forever.

![image](https://user-images.githubusercontent.com/14205420/106365910-e074c700-6338-11eb-8e8d-83058a2042ee.png)

In my opinion no one who does this is willing to lose artifacts. 
What do you propose instead? Introducing a new condition which matches build that are marked as keep forever?